src/consumer/offsetManager/index.js

```diff
@@ -140,6 +173,21 @@ module.exports = class OffsetManager {
     this.clearOffsets({ topic, partition })
   }
 
+  async commitOffsetsIfNecessary() {
+    const now = Date.now()
+
+    const timeoutReached =
+      this.autoCommitInterval != null && now >= this.lastCommit + this.autoCommitInterval
+
+    const thresholdReached =
+      this.autoCommitThreshold != null &&
+      this.countResolvedOffsets().gte(Long.fromValue(this.autoCommitThreshold))
+
+    if (timeoutReached || thresholdReached) {
+      return this.commitOffsets()
```

Comment:
If `autoCommitThreshold` and `autoCommitInterval` are both undefined, `commitOffsetsIfNecessary` will never commit offsets. That feels unintuitive, but I'm not sure if the alternative is better...

Reply:
I agree but my understanding of `necessary` in this functions is defined the by policies, so if it's not configured then it's not necessary. Like we discussed earlier, we could eventually expose `commitOffsets`.